lib/spack/spack/cmd/env.py

```diff
@@ -523,6 +524,34 @@ def env_revert(args):
     tty.msg(msg.format(manifest_file))
 
 
+def env_generate_makefile_setup_parser(subparser):
+    pass
+
+
+def env_generate_makefile(args):
+    env = ev.active_environment()
+    if not env:
+        tty.die("required env")
+    hash_to_deps_hashes = dict()
+    hash_to_spec = dict()
+    for (_, spec) in env.concretized_specs():
+        for s in spec.traverse(root=True):
+            hash_to_spec[s.dag_hash()] = s
+            hash_to_deps_hashes[s.dag_hash()] = ' '.join(
+                [dep.dag_hash() for dep in s.dependencies()])
+
+    print(".PHONY: all clean\n")
+    print("all: {}\n".format(' '.join(
+        [s.dag_hash() for (_, s) in env.concretized_specs()])))
+    for (parent, children) in hash_to_deps_hashes.items():
+        spec_format = '{name}{@version}{%compiler}{variants}{arch=architecture}'
+        print("{}: {}".format(parent, children))
+        print("\t# {}".format(hash_to_spec[parent].format(spec_format)))
+        print("\tspack -e . install --only-concrete --only=package /$@ && touch $@\n")
```

Comment:
Can the make dep files be moved to some subdirectory of the env?  This seems like it'll clutter things in the top-level env directory.  I dunno what to call the directory -- `makedeps` or something?